tests/workpool.md

```diff
@@ -0,0 +1,338 @@
+# Setting up the environment
+
+```ocaml
+# #require "eio_main";;
+# #require "eio.mock";;
+```
+
+Creating some useful helper functions
+
+```ocaml
+open Eio.Std
+
+module Workpool = Eio.Workpool
+
+let () = Eio.Exn.Backend.show := false
+
+let run fn =
+  Eio_mock.Backend.run @@ fun () ->
+  Eio_mock.Domain_manager.run @@ fun mgr ->
+  Switch.run @@ fun sw ->
+  let clock = Eio_mock.Clock.make () in
+  let sleep ms =
+    let t0 = Eio.Time.now clock in
+    let t1 = t0 +. ms in
+    traceln "Sleeping %.0f: %.0f -> %.0f" ms t0 t1;
+    Fiber.both
+      (fun () -> Eio.Time.sleep_until clock t1)
+      (fun () ->
+        Fiber.yield ();
+        Fiber.yield ();
+        Fiber.yield ();
+        Fiber.yield ();
+        Fiber.yield ();
+        Fiber.yield ();
+        Fiber.yield ();
```

Comment:
Is there a "yield until idle" function? If not, this is the precise number of yields needed...

Reply:
There isn't. It needs support in `Eio_mock.Backend`. I was thinking of having a `Backend.run_full` that provides an environment with a mock clock that auto-advances whenever the run-queue is empty.
e.g.

```ocaml
Eio_mock.Backend.run @@ fun env ->
let clock = env#clock in
...
```

Reply:
That would be fantastic to have! While reading other tests to see how mock clocks were used, I saw multiple opportunities where it could have been used.

Reply:
We also have mixed feelings about those explicit yields in the tests. A purist argument would be that it forces the writer to clearly express the semantics of a function... but in practice it is quite a headache.